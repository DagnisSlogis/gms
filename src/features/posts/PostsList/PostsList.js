import React, { PureComponent } from "react";
import PropTypes from "prop-types";
import styled from "styled-components";

import LoaderHOC from "../../../HOC/LoaderHOC";
import StackGrid from "react-stack-grid";
import PostThumbBlock from "./../PostThumbBlock/PostThumbBlock";

const PostsListBlock = styled.div`
  max-width: 1024px;
  width: 100%;
  margin: 40px auto;
  padding: 0 20px;
  overflow: hidden;
`;

type Props = {
  posts: Array<Object>,
  openPost: Function
};

class PostsList extends PureComponent<Props> {
  static propTypes = {
    posts: PropTypes.array.isRequired,
    openPost: PropTypes.func.isRequired
  };

  static defaultProps = {
    posts: []
  };

  render = () => (
    <PostsListBlock>
      <StackGrid columnWidth={328} monitorImagesLoaded={true}>
        {this.props.posts.map(post => (
          <PostThumbBlock
            key={post.id}
            post={post}
            onClick={this.props.openPost}
          />
        ))}
      </StackGrid>
    </PostsListBlock>
  );
}

export default LoaderHOC("posts")(PostsList);
