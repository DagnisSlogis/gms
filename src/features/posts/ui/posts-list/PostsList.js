import React from "react";
import PropTypes from "prop-types";
import styled from "styled-components";
import { noop } from 'lodash';
import Masonry from 'react-masonry-component';

import { PostThumbBlock } from "../post-thumb-block/PostThumbBlock";

const PostsListBlock = styled.div`
  max-width: 1024px;
  width: 100%;
  margin: 40px auto;
  padding: 0 20px;
  overflow: hidden;
`;

export const PostsList = ({ posts, openPost }) => (
  <PostsListBlock>
    <Masonry>
      {posts.map((post) => (
        <PostThumbBlock
          key={post.id}
          post={post}
          onClick={openPost}
        />
      ))}
    </Masonry>
  </PostsListBlock>
);

PostsList.propTypes = {
  posts: PropTypes.array,
  openPost: PropTypes.func
};

PostsList.defaultProps = {
  posts: [],
  openPost: noop,
};
