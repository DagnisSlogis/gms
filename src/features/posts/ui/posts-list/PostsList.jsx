import React, { PureComponent } from "react";
import PropTypes from "prop-types";
import styled from "styled-components";
import StackGrid from "react-stack-grid";

import LoaderHOC from "HOC/LoaderHOC";

import PostThumbBlock from "../post-thumb-block/PostThumbBlock";


const PostsListBlock = styled.div`
  max-width: 1024px;
  width: 100%;
  margin: 40px auto;
  padding: 0 20px;
  overflow: hidden;
`;

class PostsList extends PureComponent {
  static propTypes = {
    posts: PropTypes.array.isRequired,
    openPost: PropTypes.func
  };

  static defaultProps = {
    posts: []
  };

  render() {
    const { posts, openPost } = this.props;

    return (
      <PostsListBlock>
        <StackGrid columnWidth={328} monitorImagesLoaded >
          {posts.map((post) => (
            <PostThumbBlock
              key={post.id}
              post={post}
              onClick={openPost}
            />
          ))}
        </StackGrid>
      </PostsListBlock>
    );
  }
}

export default LoaderHOC("posts")(PostsList);
