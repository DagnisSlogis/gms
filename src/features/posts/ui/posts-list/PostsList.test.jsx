// jest PostsList
import React from "react";
import { shallow } from "enzyme";
import PostsList from "./PostsList";
import "jest-styled-components";

describe("<PostsList/>", () => {
  describe("Has posts in posts prop", () => {
    let wrapper;
    const posts = [{ id: 1, title: "Post 1" }, { id: 2, title: "Post 2" }];

    beforeEach(() => {
      wrapper = shallow(<PostsList posts={posts}/>);
    });

    it("renders component", () => {
      expect(wrapper.length).toEqual(1);
    });

    it("snapshot test", () => {
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe("Has no posts in posts prop", () => {
    let wrapper;
    const posts = [];

    beforeEach(() => {
      wrapper = shallow(<PostsList posts={posts}/>);
    })

    it("renders component", () => {
      expect(wrapper.length).toEqual(1);
    });

    it("renders isLoading text", () => {
      expect(wrapper.text()).toContain("is Loading");
    });

    it("snapshot test", () => {
      expect(wrapper).toMatchSnapshot();
    });
  });
});
