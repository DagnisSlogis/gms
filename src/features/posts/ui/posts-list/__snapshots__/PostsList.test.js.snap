// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PostsList/> Has no posts in posts prop snapshot test 1`] = `
<div>
  is Loading
</div>
`;

exports[`<PostsList/> Has posts in posts prop snapshot test 1`] = `
<PostsList
  posts={
    Array [
      Object {
        "id": 1,
        "title": "Post 1",
      },
      Object {
        "id": 2,
        "title": "Post 2",
      },
    ]
  }
/>
`;
